modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
+
+    returns:  The longest common prefix path of all the input paths
+    */
+    proc commonPrefix(paths: []): string {
+
+    var prefix: string = "";    // result string
+    var inputLength = paths.size;   // size of input array
+    var firstPath = paths[1];
+    var delimiter: string;
+    var cnt: int = 1;
+    var pos: int ;    // rightmost index of common prefix
+    var minPathLength: int ;
+
+    // if input is empty, return empty string.
+    // if input is just one string, return that string as longest common prefix path.
+
+    if inputLength == 0 then {
+      return prefix;
+    } else if inputLength == 1 then{
+      return firstPath;
+    }
+
+    // finding delimiter to split the paths.
+
+    if firstPath.find("\\", 1..firstPath.length) == 0 then {
+      delimiter = "/";
+    } else {
+      delimiter = "\\";
+    }
+    cnt = firstPath.count(delimiter);
+
+    var prefixArray:[1..cnt+1] string;    // array of resultant prefix string
+    prefixArray = firstPath.split(delimiter, -1, false);
+
+    pos = prefixArray.size;
+    minPathLength = prefixArray.size;
+
+    for i in 2..inputLength do {
+
+      cnt = paths[i].count(delimiter);
+      var tempArray:[1..cnt+1] string;    // temporary array storing the current path under consideration
+      tempArray = paths[i].split(delimiter, -1, false);
+
+      var minimum = min(prefixArray.size, cnt+1);
```

Comment:
In answer to your question about zippering, it looks like we only support zippering of different lengths when one is unbounded (see [here](https://chapel-lang.org/docs/latest/users-guide/base/zip.html?highlight=zippering#zippered-iterand-size-matching) for more information on what that means).  The documentation mentions being able to send in a size argument to the iterator, but I'm not seeing an example of how to do that quickly.  This seems like a reasonable question to ask either in the gitter/irc chatroom or on Stack Overflow, as someone else probably knows the answer to that.